### PR TITLE
Init {Config,Permissions}Resource through CollectionRegistry

### LINKED ISSUE #2669: Exception thrown during startup
**Affected module**
Does it impact the UI, backend, or Ingestion Framework?
Backend

**Describe the bug**
The following exception is thrown when the system starts up:
```
WARN  [2022-02-08 05:52:59,588] org.openmetadata.catalog.resources.CollectionRegistry: Failed to create resource for class org.openmetadata.catalog.resources.permissions.PermissionsResource {}
! java.lang.NoSuchMethodException: org.openmetadata.catalog.resources.permissions.PermissionsResource.<init>()
! at java.base/java.lang.Class.getConstructor0(Class.java:3349)
! at java.base/java.lang.Class.getConstructor(Class.java:2151)
! at org.openmetadata.catalog.resources.CollectionRegistry.createResource(CollectionRegistry.java:204)
! at org.openmetadata.catalog.resources.CollectionRegistry.registerResources(CollectionRegistry.java:129)
! at org.openmetadata.catalog.CatalogApplication.registerResources(CatalogApplication.java:233)
! at org.openmetadata.catalog.CatalogApplication.run(CatalogApplication.java:119)
! at org.openmetadata.catalog.CatalogApplication.run(CatalogApplication.java:71)
! at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:59)
! at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:98)
! at io.dropwizard.testing.DropwizardTestSupport.startIfRequired(DropwizardTestSupport.java:326)
! at io.dropwizard.testing.DropwizardTestSupport.before(DropwizardTestSupport.java:225)
! at io.dropwizard.testing.junit5.DropwizardAppExtension.before(DropwizardAppExtension.java:195)
! at io.dropwizard.testing.junit5.DropwizardExtensionsSupport.beforeAll(DropwizardExtensionsSupport.java:97)
! at io.dropwizard.testing.junit5.DropwizardExtensionsSupport.beforeAll(DropwizardExtensionsSupport.java:83)
! at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$10(ClassBasedTestDescriptor.java:381)
! at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
! at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:381)
! at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:205)
! at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:80)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
! at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
! at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
! at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
! at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
! at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
```
**To Reproduce**
Just start the system

**Expected behavior**
Exception now thrown

**Version:**
 - OpenMetadata version: [e.g. 0.8]

**Additional context**
@mithmatt, you need to add the following tag as shown below from TableResource.java:
```
@Collection(name = "tables")
```
With this, you don't need to register this resource manually. It is automatically picked up by `CollectionRegistry.java`
Can you please add similar annotation to `ConfigResource.java` and remove manually registering also? There is an exception for that as well.

### PR DESCRIPTION
**Pull request by mithmatt**
### Describe your changes :

I worked on this to fix #2669

#
### Type of change :

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

 Backend: @sureshms @harshach

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/permissions/PermissionsResource.java [hunk 1]
```diff
@@ -37,7 +38,7 @@
 public class PermissionsResource {
   private final Authorizer authorizer;
```
[the remark this answers is not in the document]
Comment (mithmatt) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/permissions/PermissionsResource.java:39]: The annotation already exists for both
```
@Collection(name = "permissions")
public class PermissionsResource {
```

However, the CollectionRegistry did not find the constructor method with the argument signature (CollectioDAO, Authorizer) it was looking for to initialize the Resources

**Review by vivekratnavel — APPROVED** [review]
+1 LGTM

**Comment by harshach** [comment]
@mithmatt since you are looking into this, we are trying to register ConfigResource twice https://github.com/open-metadata/OpenMetadata/blob/main/catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplication.java#L207 one through this explicit call and one will be part of registerResources which is the reason it throws an error there . Lets keep one way of registering config resource remove the other

**Comment by mithmatt** [comment]
> @mithmatt since you are looking into this, we are trying to register ConfigResource twice https://github.com/open-metadata/OpenMetadata/blob/main/catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplication.java#L207 one through this explicit call and one will be part of registerResources which is the reason it throws an error there . Lets keep one way of registering config resource remove the other

Yes I saw that and removed the one in CatalogApplication as part of this PR.

With this everything goes through CollectionRegistry

**Pull request merged**